frontend/src/modules/search/components/SearchBox.js

```diff
@@ -136,31 +150,45 @@ export class SearchBoxBase extends React.Component<InternalProps, State> {
         });
     }
 
-    applySingleFilter = (filter: string, callback?: () => void) => {
+    applySingleFilter = (filter: string, type: string, callback?: () => void) => {
         const statuses = this.getInitialStatuses();
         const extras = this.getInitialExtras();
+        const tags = this.getInitialTags();
 
         if (filter !== 'all') {
-            if (filter in statuses) {
-                statuses[filter] = true;
-            }
-            else if (filter in extras) {
-                extras[filter] = true;
+            switch (type) {
+                case 'statuses':
+                    statuses[filter] = true;
+                    break;
+                case 'extras':
+                    extras[filter] = true;
+                    break;
+                case 'tags':
+                    tags[filter] = true;
```

Comment:
You want another `break` here and a `default:` otherwise flow will complain. And I believe rightfully so, it's better to always have a `default` clause in your switch, even if it does nothing.

Reply:
Good call. Thew build even failed. Updated.